core/chains/evm/txm/txm.go

```diff
@@ -0,0 +1,443 @@
+	if err == nil {
+		t.lggr.Infow("Created transaction", "tx", tx)
+	}
+	return
+}
+
+func (t *Txm) Trigger(address common.Address) {
+	if !t.IfStarted(func() {
+		triggerCh, exists := t.triggerCh[address]
+		if !exists {
+			return
+		}
+		triggerCh <- struct{}{}
+	}) {
+		t.lggr.Error("Txm unstarted")
+	}
+}
+
+func (t *Txm) Abandon(address common.Address) error {
+	return t.txStore.AbandonPendingTransactions(context.TODO(), address)
+}
+
+func (t *Txm) getNonce(address common.Address) uint64 {
+	t.nonceMapMu.Lock()
+	defer t.nonceMapMu.Unlock()
+	return t.nonceMap[address]
+}
+
+func (t *Txm) setNonce(address common.Address, nonce uint64) {
+	t.nonceMapMu.Lock()
+	t.nonceMap[address] = nonce
+	defer t.nonceMapMu.Unlock()
+}
+
+func newBackoff(minDuration time.Duration) backoff.Backoff {
+	return backoff.Backoff{
+		Min:    minDuration,
+		Max:    1 * time.Minute,
+		Jitter: true,
+	}
+}
+
+func (t *Txm) broadcastLoop(address common.Address, triggerCh chan struct{}) {
+	defer t.wg.Done()
+	ctx, cancel := t.stopCh.NewCtx()
+	defer cancel()
+	broadcastWithBackoff := newBackoff(1 * time.Second)
+	var broadcastCh <-chan time.Time
+
+	for {
+		start := time.Now()
+		bo, err := t.broadcastTransaction(ctx, address)
+		if err != nil {
+			t.lggr.Errorw("Error during transaction broadcasting", "err", err)
+		} else {
+			t.lggr.Debug("Transaction broadcasting time elapsed: ", time.Since(start))
+		}
+		if bo {
+			broadcastCh = time.After(broadcastWithBackoff.Duration())
+		} else {
+			broadcastWithBackoff.Reset()
+			broadcastCh = time.After(utils.WithJitter(broadcastInterval))
+		}
+		select {
+		case <-ctx.Done():
+			return
+		case <-triggerCh:
+			continue
+		case <-broadcastCh:
+			continue
+		}
+	}
+}
+
+func (t *Txm) backfillLoop(address common.Address) {
+	defer t.wg.Done()
+	ctx, cancel := t.stopCh.NewCtx()
+	defer cancel()
+	backfillWithBackoff := newBackoff(t.config.BlockTime)
+	backfillCh := time.After(utils.WithJitter(t.config.BlockTime))
+
+	for {
+		select {
+		case <-ctx.Done():
+			return
+		case <-backfillCh:
+			start := time.Now()
+			bo, err := t.backfillTransactions(ctx, address)
+			if err != nil {
+				t.lggr.Errorw("Error during backfill", "err", err)
+			} else {
+				t.lggr.Debug("Backfill time elapsed: ", time.Since(start))
+			}
+			if bo {
+				backfillCh = time.After(backfillWithBackoff.Duration())
+			} else {
+				backfillWithBackoff.Reset()
+				backfillCh = time.After(utils.WithJitter(t.config.BlockTime))
+			}
+		}
+	}
+}
+
+func (t *Txm) broadcastTransaction(ctx context.Context, address common.Address) (bool, error) {
+	for {
+		_, unconfirmedCount, err := t.txStore.FetchUnconfirmedTransactionAtNonceWithCount(ctx, 0, address)
+		if err != nil {
+			return false, err
+		}
+
+		// Optimistically send up to 1/3 of the maxInFlightTransactions. After that threshold, broadcast more cautiously
+		// by checking the pending nonce so no more than maxInFlightTransactions/3 can get stuck simultaneously i.e. due
```

Comment:
Not really feedback but curious why `maxInFlightTransactions/3`?

Reply:
A third of the `maxInFlightTransactions` seemed like a reasonable number of transactions to be potentially stuck before "paying" for an RPC request.

Reply:
No, that was on purpose. I've never seen this value change, so I want to remove the config and only add it if needed.

Reply:
In that case, 1/3 of `maxInFlightTransactions` is also a static number. Could we set an explicit limit instead? Think it might be clearer if we say we broadcast more carefully after 5 transactions.